enumerables.rb

```diff
@@ -0,0 +1,293 @@
+      arr_str = map { |i| i.to_s }
+      arr_str.my_each { |elmt| counter += 1 if elmt.match?(arg) }
+
+    # Not there a block or an argument?
+    elsif arg.nil?
+      my_each { |elmt| counter += 1 if elmt != false && !elmt.nil? }
+
+    # My_none? will be used to find out if the collection is not made of a specific object
+    else
+      my_each { |elmt| counter += 1 if elmt == arg }
+    end
+    counter.zero?
+  end
+end
+
+module Enumerable
+  # 7.- ---- my_count Method ----
+
+  def my_count(*arg)
+    count = 0
+
+    if block_given?
+      my_each do |elmt|
+        count += 1 if yield(elmt)
+      end
+    elsif arg.empty?
+      count = size
+    else
+      my_each do |elmt|
+        count += 1 if elmt == arg[0]
+      end
+    end
+    count
+  end
+
+  # 8.- ---- my_map Method ----
+
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    new_arr = []
+
+    if self.class == Array or self.class == Range
+      my_each { |i| new_arr.push(yield(i)) }
+    elsif self.class == Hash
+      my_each { |key, value| new_arr.push(yield(key, value)) }
+    end
+    new_arr
+  end
+end
+
+module Enumerable
+  # 9.- ---- my_inject Method ----
+
+  def my_inject(*args)
+    memo = to_a[0]
+
+    # Case 1: arguments are given but not a block
+    if !args.empty? && !block_given?
+      # There are an initial value and a symbol which indicates the operation to perform
+      if (args[0].is_a? Numeric or args[0].class == String) && args[1].class == Symbol
+        memo = args[0]
+        if args[1] == :+
+          my_each { |elmt| memo += elmt }
+        elsif args[1] == :-
+          my_each { |elmt| memo -= elmt }
+        elsif args[1] == :*
+          my_each { |elmt| memo *= elmt }
+        elsif args[1] == :/
+          my_each { |elmt| memo /= elmt }
+        elsif args[1] == :**
+          my_each { |elmt| memo **= elmt }
+        elsif args[1] == :%
+          my_each { |elmt| memo %= elmt }
+        end
+
+      elsif args.size == 1 && args[0].class == Symbol # There's only the operation's symbol
+        if args[0] == :+
+          self[1..-1].my_each { |elmt| memo += elmt }
+        elsif args[0] == :-
+          self[1..-1].my_each { |elmt| memo -= elmt }
+        elsif args[0] == :*
+          self[1..-1].my_each { |elmt| memo *= elmt }
+        elsif args[0] == :/
+          self[1..-1].my_each { |elmt| memo /= elmt }
+        elsif args[0] == :**
+          self[1..-1].my_each { |elmt| memo **= elmt }
+        elsif args[0] == :%
+          self[1..-1].my_each { |elmt| memo %= elmt }
+        end
+
+      end
+
+      # Case 2: an initial value is given as an argument along with a block
+    elsif (args[0].is_a? Numeric or args[0].class == String) && block_given?
+      memo = args[0]
+      my_each { |elmt| memo = yield(memo, elmt) }
+
+      # Case 3: only a block is given
+    elsif args.empty? && block_given?
+      self[1..-1].my_each { |elmt| memo = yield(memo, elmt) }
+    end
+
+    memo
+  end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving the wrong result
![image](https://user-images.githubusercontent.com/52209067/78953742-1ccbf380-7a97-11ea-9011-c1cf375ace99.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
<img width="662" alt="9" src="https://user-images.githubusercontent.com/58504655/98773400-ac10ff00-240a-11eb-913c-95c8aa0cfe36.PNG"> 



- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)

Reply:
Understood. Again i made few changes to #my_inject method and tested in Repl.it. Here the results:

- When looking for #my_inject to return the same as Ruby's #inject 
![image](https://user-images.githubusercontent.com/63531796/98877575-6a9e4500-2446-11eb-87ad-e06fe7029a01.png)